metagraph/plugins/numpy/types.py

```diff
@@ -433,3 +468,19 @@ def assert_equal(
                     assert np.isclose(d1, d2, rtol=rel_tol, atol=abs_tol).all().all()
                 else:
                     assert (d1 == d2).all().all()
+
+
+class NumpyNodeEmbedding(NodeEmbeddingWrapper, abstract=NodeEmbedding):
+    def __init__(self, matrix, nodes=None):
+        super().__init__(matrix, nodes)
+        self._assert_instance(matrix, NumpyMatrix)
+        if nodes is not None:
+            self._assert_instance(nodes, NumpyNodeMap)
+            self._assert(
+                len(nodes) == matrix.value.shape[0],
+                f"Node count ({len(nodes)}) and matrix row count ({matrix.value.shape[0]})do not match.",
+            )
+
+    def copy(self):
+        nodes = self.nodes if self.nodes is None else self.nodes.copy()
+        return NumpyNodeEmbedding(self.edges.matrix(), nodes=nodes)
```

Comment:
What is `self.edges`? Is this a copy-paste error from CompositeGraph?

Reply:
Yes, that's a copy and paste error. I'll fix it.

Reply:
https://github.com/metagraph-dev/metagraph/pull/133/commits/629feccbe10a02523619dcfb25f4e70d89e3428c intends to fix the copy-and-paste error.